src/main/java/seedu/address/model/person/Faculty.java

```diff
@@ -0,0 +1,83 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.stream.Stream;
+
+/**
+ * Represents a Person's faculty in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidFaculty(String)}
+ */
+public class Faculty {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Only the following faculties exists: " + getFacultyEnumAsString();
+
+    public final String studentFaculty;
+
+    /**
+     * Constructs a {@code Faculty}.
+     *
+     * @param faculty A valid faculty.
+     */
+    public Faculty(String faculty) {
+        requireNonNull(faculty);
+        checkArgument(isValidFaculty(faculty), MESSAGE_CONSTRAINTS);
+        studentFaculty = faculty.toUpperCase();
```

Comment:
I think maybe can stick to one coding standard regarding the assignment to the attributes? in the Person class there is a reference to the object through "this" but in some classes(Faculty, CovidStatus etc) there isn't. Just a small issue but I think we should stick to one standard throughout our code.

Reply:
Can discuss about this in next meeting because the name of the class attributes is the same name as the constructor parameters, therefore it requires the 'this' keyword